tv_apps/smart-system/js/app_install_manager.js

```diff
@@ -79,6 +94,21 @@ var AppInstallManager = {
 
     // TODO: write an integration test for pressing home.
     window.addEventListener('home', this.hideAllDialogs.bind(this));
+
+    window.addEventListener('appopening', (evt) => {
+      if (evt.detail.url.indexOf('app-deck.gaiamobile.org') !== -1) {
+        this.clearPreviewApps();
+      }
+    });
+
+    window.addEventListener('previewterminated', (evt) => {
+      this.clearPreviewApps();
+    });
+  },
+
+  isMarketPlaceAppActive: function ai_isMarketPlaceAppActive() {
+    var activeApp = AppWindowManager.getActiveApp() || { manifestURL: '' };
+    return (activeApp.manifestURL.indexOf(this.marketplaceURLPattern) !== -1);
```

Comment:
I'd prefer `return activeApp && (activeApp.mani...... !== -1);` so you don't need to mock a manifest object in previous line.